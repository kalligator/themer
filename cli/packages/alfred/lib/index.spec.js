const { render, renderInstructions } = require('./index');
const { colors } = require('../../colors-default');

describe('themer Alfred.app theme template', () => {
  it('should render properly', async () => {
    const files = await Promise.all(render(colors));
    expect(files.length).toBe(2);
    files.forEach((file) => {
      expect(file.contents.toString('utf8')).toMatchSnapshot();
    });
  });
  it('should provide installation instructions', async () => {
    const files = await Promise.all(render(colors));
    const instructions = renderInstructions(files.map(({ name }) => name));
    expect(instructions).toMatchSnapshot();
  });
});
