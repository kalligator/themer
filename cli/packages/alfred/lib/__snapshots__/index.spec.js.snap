// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`themer Alfred.app theme template should provide installation instructions 1`] = `
"
Simply open the files to import them into Alfred. Either double-click them in Finder or use the terminal:

    open 'themer-alfred-dark.alfredappearance'
    open 'themer-alfred-light.alfredappearance'
"
`;

exports[`themer Alfred.app theme template should render properly 1`] = `
"{
  \\"alfredtheme\\": {
    \\"result\\": {
      \\"textSpacing\\": 6,
      \\"subtext\\": {
        \\"size\\": 12,
        \\"colorSelected\\": \\"#474247FF\\",
        \\"font\\": \\"System Light\\",
        \\"color\\": \\"#A29DA3FF\\"
      },
      \\"shortcut\\": {
        \\"size\\": 16,
        \\"colorSelected\\": \\"#474247FF\\",
        \\"font\\": \\"System\\",
        \\"color\\": \\"#E0DCE0FF\\"
      },
      \\"backgroundSelected\\": \\"#26C99EFF\\",
      \\"text\\": {
        \\"size\\": 18,
        \\"colorSelected\\": \\"#282629FF\\",
        \\"font\\": \\"System Light\\",
        \\"color\\": \\"#FFFCFFFF\\"
      },
      \\"iconPaddingHorizontal\\": 6,
      \\"paddingVertical\\": 6,
      \\"iconSize\\": 36
    },
    \\"search\\": {
      \\"paddingVertical\\": 8,
      \\"background\\": \\"#6560667F\\",
      \\"spacing\\": 10,
      \\"text\\": {
        \\"size\\": 22,
        \\"colorSelected\\": \\"#66BFFFFF\\",
        \\"font\\": \\"System\\",
        \\"color\\": \\"#FFFCFFFF\\"
      },
      \\"backgroundSelected\\": \\"#F553BFFF\\"
    },
    \\"window\\": {
      \\"color\\": \\"#282629CC\\",
      \\"paddingHorizontal\\": 10,
      \\"width\\": 560,
      \\"borderPadding\\": 0,
      \\"borderColor\\": \\"#28262900\\",
      \\"blur\\": 15,
      \\"roundness\\": 2,
      \\"paddingVertical\\": 10
    },
    \\"credit\\": \\"Themer\\",
    \\"separator\\": {
      \\"color\\": \\"#28262900\\",
      \\"thickness\\": 0
    },
    \\"scrollbar\\": {
      \\"color\\": \\"#FFD24AFF\\",
      \\"thickness\\": 2
    },
    \\"name\\": \\"Themer Dark\\"
  }
}"
`;

exports[`themer Alfred.app theme template should render properly 2`] = `
"{
  \\"alfredtheme\\": {
    \\"result\\": {
      \\"textSpacing\\": 6,
      \\"subtext\\": {
        \\"size\\": 12,
        \\"colorSelected\\": \\"#E0DCE0FF\\",
        \\"font\\": \\"System Light\\",
        \\"color\\": \\"#847E85FF\\"
      },
      \\"shortcut\\": {
        \\"size\\": 16,
        \\"colorSelected\\": \\"#E0DCE0FF\\",
        \\"font\\": \\"System\\",
        \\"color\\": \\"#474247FF\\"
      },
      \\"backgroundSelected\\": \\"#1FC598FF\\",
      \\"text\\": {
        \\"size\\": 18,
        \\"colorSelected\\": \\"#FFFCFFFF\\",
        \\"font\\": \\"System Light\\",
        \\"color\\": \\"#282629FF\\"
      },
      \\"iconPaddingHorizontal\\": 6,
      \\"paddingVertical\\": 6,
      \\"iconSize\\": 36
    },
    \\"search\\": {
      \\"paddingVertical\\": 8,
      \\"background\\": \\"#C1BCC27F\\",
      \\"spacing\\": 10,
      \\"text\\": {
        \\"size\\": 22,
        \\"colorSelected\\": \\"#53A6E1FF\\",
        \\"font\\": \\"System\\",
        \\"color\\": \\"#282629FF\\"
      },
      \\"backgroundSelected\\": \\"#EE4EB8FF\\"
    },
    \\"window\\": {
      \\"color\\": \\"#FFFCFFCC\\",
      \\"paddingHorizontal\\": 10,
      \\"width\\": 560,
      \\"borderPadding\\": 0,
      \\"borderColor\\": \\"#FFFCFF00\\",
      \\"blur\\": 15,
      \\"roundness\\": 2,
      \\"paddingVertical\\": 10
    },
    \\"credit\\": \\"Themer\\",
    \\"separator\\": {
      \\"color\\": \\"#FFFCFF00\\",
      \\"thickness\\": 0
    },
    \\"scrollbar\\": {
      \\"color\\": \\"#EEBA21FF\\",
      \\"thickness\\": 2
    },
    \\"name\\": \\"Themer Light\\"
  }
}"
`;
